pkg/cache/task.go

```diff
@@ -440,6 +460,23 @@ func (task *Task) beforeTaskAllocated(event *fsm.Event) {
 }
 
 func (task *Task) postTaskBound(event *fsm.Event) {
+	if task.pluginMode {
+		// when the pod is scheduling by yunikorn, it is moved to the default-scheduler's
+		// unschedulable queue, if nothing changes, the pod will be staying in the unschedulable
+		// queue for unschedulableQTimeInterval long (default 1 minute). hence, we are updating
+		// the pod status explicitly, when there is a status change, the default scheduler will
+		// move the pod back to the active queue immediately.
+		podCopy := task.pod.DeepCopy()
```

Comment:
Can we reason about the performance effects of this copy? If we do this all the time (and that's what seems to be happening), we end up creating extra objects which affects GC time. Maybe not a big deal, just something to think about.

Reply:
This cost is negligible, because the copy happens very infrequently (typically once per pod), when YuniKorn has decided to allocate the pod. We update the pod status once at this point to force the default scheduler logic to re-consider the pod for scheduling. It will very shortly call PreFilter() / Filter() and push the Pod through the scheduling cycle.